File: loops.js
// Write a program to display 'n' terms of natural number and their sum.

let n = +prompt("Enter the number of terms");
let sum = 0;

for (let i = 1; i <= n; i++) {
    sum = sum + i;
    console.log(i);
}
console.log("The sum of natural numbers is " + sum);

// Write a program to display the cube of the number up to given integer.

let n = +prompt("Enter the number");
let sum = 0;

while (n >= 0) {
    sum = sum + n * n * n;
    n--;
}
console.log("The sum of cubes is " + sum);

for (let i = 1; i <= n; i++) {
    sum = sum + i * i * i;
    console.log(i * i * i);
}
console.log("The sum of cubes is " + sum);

// Write a program to display the multiplication table of a given integer

let n = +prompt("Enter the number");
let sum = 0;

while (n > 0) {
    sum = sum + n;
    console.log(n);
    n--;
}
console.log("The sum of multiplication table is " + sum);

for (let i = 1; i <= 10; i++) {
    sum = sum + n * i;
    console.log(n + " * " + i + " = " + n * i);
}
console.log("The sum of multiplication is " + sum);

// Write a program to sum all the digits of an integer.

let n = +prompt("Enter the number");
let sum = 0;

while (n > 0) {
    sum = sum + n % 10;
    n = Math.floor(n / 10);
}
console.log("The sum of digits is " + sum);

for (let i = 1; i <= n; i++) {
    sum = sum + i;
    console.log(i);
}
console.log("The sum of digits is " + sum);

// Write program to calculate the factorial of a given number

let n = +prompt("Enter the number");
let sum = 1;

while (n > 0) {
    sum = sum * n;
    n--;
}
console.log("The factorial of " + n + " is " + sum);

for (let i = 1; i <= n; i++) {
    sum = sum * i;
    console.log(i);
}
console.log("The factorial of " + n + " is " + sum);

// Write a program to display the 'n' terms of square natural number and their sum.

let n = +prompt("Enter the number");
let sum = 0;

while (n > 0) {
    sum = sum + n * n;
    console.log(n * n);
    n--;
}
console.log("The sum of squares is " + sum);

for (let i = 1; i <= n; i++) {
    sum = sum + i * i;
    console.log(i * i);
}
console.log("The sum of squares is " + sum);

// Write a program to convert a decimal number into binary.

let n = +prompt("Enter the number");
let sum = 0;

while (n > 0) {
    sum = sum + n % 2;
    n = Math.floor(n / 2);
}
console.log("The binary number is " + sum);

for (let i = 1; i <= n; i++) {
    sum = sum + i % 2;
    console.log(i % 2);
}
console.log("The binary number is " + sum);

File: loops.py
# Print first 10 Natural numbers.

i = 1
while i <= 10:
    print(i)
    i += 1

# Write a program to calculate sum of number from 1 to n.

s = 0
n = int(input("Enter a number: "))
for i in range(1, n+1):
    s += i
print("Sum of first", n, "natural numbers is:", s)

# Write a program to print multiplication table of a given number.

n = 2
for i in range(1, 11):
    print(n, "*", i, "=", n*i)

# Count the total number of digits in a number.

n = 57830
count = 0
while n > 0:
    n = n // 10
    count += 1
print("Total number of digits in", 57830, "is:", count)

# Print list in reverse order using a loop.

list = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
for i in range(len(list)-1, -1, -1):
    print(list[i])

# Display numbers from -10 to -1 using for loop.

for num in range (-10, 0):
    print(num)

# Use else block to display a message “Done” after successful execution of for loop.

for i in range(5):
    print(i)
else:
    print("Done")

    